geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/wancommand/ClusterConfigStartStopPauseAndResumeGatewaySenderOperationDUnitTest.java

```diff
@@ -0,0 +1,716 @@
+    // start threads
+    thread.start();
+    thread1.start();
+    thread.join();
+    thread1.join();
+
+    verifyGatewaySenderState(false, false);
+    if (isParallel.equals("true")) {
+      server1Site2.invoke(() -> checkQueueSize("ln", keysQueued.size()));
+    }
+    executeGfshCommand(CliStrings.START_GATEWAYSENDER);
+    verifyGatewaySenderState(true, false);
+
+    // Check that data is sent over the gateway-sender
+    server1Site2.invoke(() -> checkQueueSize("ln", 0));
+    clientSite1.invoke(() -> checkDataAvailable(keysQueued));
+    clientSite1.invoke(() -> checkDataNotAvailable(keysNotQueued));
+  }
+
+  /**
+   * This test case verifies that the server during recovery of gateway-sender in stopped state
+   * stores events in tmpDroppedEvents queue, and handles them after sender has been recovered.
+   */
+  @Test
+  public void testSubscriptionQueueWanTrafficWhileServerIsRestarted() throws Exception {
+    configureSites(true, "PARTITION_REDUNDANT_PERSISTENT", "1", "true");
+
+    List<MemberVM> allMembers = new ArrayList<>();
+    allMembers.add(server1Site2);
+    allMembers.add(server2Site2);
+
+    executeGfshCommand(CliStrings.STOP_GATEWAYSENDER);
+    verifyGatewaySenderState(false, false);
+
+    Set<String> keys1 = clientSite2.invoke(() -> doPutsInRange(0, 500));
+    clientSite2.invoke(() -> checkDataAvailable(keys1));
+    server1Site2.invoke(() -> checkQueueSize("ln", 0));
+
+    MemberVM startServer = clusterStartupRule.startServerVM(9, locatorSite2.getPort());
+
+    // perform rebalance operation to redistribute primaries on running servers
+    String command = new CommandStringBuilder(CliStrings.REBALANCE)
+        .getCommandString();
+    gfsh.executeAndAssertThat(command).statusIsSuccess();
+
+    startServer.stop(false);
+
+    Thread thread = new Thread(
+        () -> clientSite2.invoke(() -> doPutsInRange(0, 15000)));
+    thread.start();
+
+    startServer = clusterStartupRule.startServerVM(9, locatorSite2.getPort());
+
+    allMembers.add(startServer);
+
+    for (MemberVM member : allMembers) {
+      member.invoke(() -> {
+        // test that non of the events are stored in primary and secondary buckets
+        testLocalQueueIsEmpty("ln");
+        // check that tmpDroppedEvent queue has been drained after it recovered in stopped state
+        verifyTmpDroppedEventSize("ln", 0);
+      });
+    }
+  }
+
+  public static void verifyTmpDroppedEventSize(String senderId, int size) {
+    GatewaySender sender = getGatewaySender(senderId);
+
+    AbstractGatewaySender ags = (AbstractGatewaySender) sender;
+    await().untilAsserted(() -> assertEquals("Expected tmpDroppedEvents size: " + size
+        + " but actual size: " + ags.getTmpDroppedEventSize(), size, ags.getTmpDroppedEventSize()));
+  }
+
+  public static void testLocalQueueIsEmpty(String senderId) {
+    await()
+        .untilAsserted(() -> localQueueSize(senderId));
+  }
+
+  public static void localQueueSize(String senderId) {
```

Comment:
I would change the name of this method to something like:
`assertThatLocalQueueSizeIsZero`